### PR TITLE
chore(edit-content): Use DotContentletThumbnail Component in Binary File

### LINKED ISSUE #26907: Edit Content: Thumbnail in the Binary Preview Punchlist
### Parent
#25445 

### User Story
As a content creator, I want to be able to render PDF thumbnails in the binary preview, so I can easily identify PDF files.

```[tasklist]
### Tasks
- [x] Use the `/dA/<%=inode%>/<%=field.getVelocityVarName()%>/500mw/20q/<%=fileName%>` URL to render ALL image thumbnails
- [x] The thumbnail should be properly sized and rendered
- [x] The thumbnail should accurately reflect the content of the PDF
- [x] Label to show image width and height as W x H, not W,H
- [x] font size in monaco editor should  compute to 14px, not 12px
- [x] File size (Bytes) should be converted to kb/mb/gb based on the file size (meaning should be human friendly). Also bytes are not working locally for me. Is anyone else seeing this issue? (we can do a pipe)
- [x] To maintain consistency, use the `dot-content let-thumbnail` component in the Binary field
- [x] Enable editing when the user clicks on editable files.
- [x] Close Monaco Editor on `ESC` without closing the entire dialog
- [x] Enable save on Binary Field on the old Edit Content
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- The PDF file must be properly formatted
- Appropriate memory resources should be available

### Quality Assurance Notes & Workarounds
- Thumbnail should be properly rendered on all major browsers

### Sub-Tasks & Estimates
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### changes

- To maintain consistency, use the `dot-content let-thumbnail` component in the Binary field.
- Enable editing when the user clicks on editable files.
- Close Monaco Editor on `ESC` without closing the entire dialog

#### Thumbnail

https://github.com/dotCMS/core/assets/72418962/1aee3fd6-b783-44f3-b473-7ac6a589b1f2

#### File Editing

https://github.com/dotCMS/core/assets/72418962/c01dfcb9-f6dd-418c-b41c-b1fc47b15db6

### CODE REVIEW

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/dotcms-webcomponents/src/elements/dot-contentlet-thumbnail/dot-contentlet-thumbnail.tsx [hunk 1]
```diff
@@ -87,6 +96,7 @@ export class DotContentletThumbnail {
     }
 
     private shouldShowVideoThumbnail() {
+        console.log(this.contentlet);
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/dotcms-webcomponents/src/elements/dot-contentlet-thumbnail/dot-contentlet-thumbnail.tsx:99]: Thanks!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-preview/dot-binary-field-preview.component.html [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
         <div class="preview-image__container">
-            <dot-content-thumbnail
-                [url]="file.url"
-                [inode]="file.inode"
-                [name]="file.name"
-                [contentType]="file.contentType"
-                [titleImage]="file.titleImage"
-                [objectFit]="objectFit"
-                [iconSize]="'48px'"></dot-content-thumbnail>
+            <ng-container *ngIf="tempFile; else contentletThumbnail">
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-preview/dot-binary-field-preview.component.html:12]: is necessary wrap in ng-container here?

File: core-web/libs/dotcms-webcomponents/src/elements/dot-video-thumbnail/dot-video-thumbnail.tsx [hunk 3]
```diff
@@ -31,10 +39,14 @@ export class DotVideoThumbnail {
...
-                        <img src={this.src} alt={title} />
-                    </div>
+                {this.playable ? (
+                    <video src={`/dA/${this.contentlet.inode}`} controls></video>
+                ) : (
+                    this.src && (
+                        <div class={`thumbnail ${cssClass}`} style={bgImage}>
+                            <img src={this.src} alt={title} />
+                        </div>
+                    )
```
Comment (fmontes) [inline core-web/libs/dotcms-webcomponents/src/elements/dot-video-thumbnail/dot-video-thumbnail.tsx:49]: Is no src, should we add an icon?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-preview/dot-binary-field-preview.component.html (hunk 2 above)
Comment (fmontes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-preview/dot-binary-field-preview.component.html]: Looks like not.

File: core-web/libs/ui/src/lib/components/dot-temp-file-thumbnail/dot-temp-file-thumbnail.component.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    inode: string;
+    name: string;
+    contentType: string;
+    tempUrl?: string;
+    iconSize?: string;
+    titleImage?: string;
+}
+
+@Component({
+    selector: 'dot-temp-file-thumbnail',
```
Comment (fmontes) [inline core-web/libs/ui/src/lib/components/dot-temp-file-thumbnail/dot-temp-file-thumbnail.component.ts:32]: Where do we use this?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

File: core-web/libs/dotcms-webcomponents/src/components.d.ts [hunk 5]
```diff
@@ -352,6 +352,7 @@ export namespace Components {
         "cover": boolean;
         "height": string;
         "iconSize": string;
+        "playableVideo": boolean;
```
Comment (oidacra) [inline core-web/libs/dotcms-webcomponents/src/components.d.ts:355]: this file is generated?

**Review by oidacra — APPROVED** [review]

**Pull request merged**